code/game/objects/structures/mannequin.dm

```diff
@@ -0,0 +1,1386 @@
+
+	else if(istype(W, /obj/item/weapon/weldingtool) && (user.a_intent == I_HELP))
+		if(locked)
+			to_chat(user, "<span class='warning'>You need to open the shield before you can fix the mannequin.</span>")
+		else
+			if(health >= maxHealth)
+				to_chat(user, "<span class='warning'>Nothing to fix here!</span>")
+				return
+			var/obj/item/weapon/weldingtool/WT = W
+			if(WT.remove_fuel(5))
+				playsound(loc, 'sound/items/Welder.ogg', 50, 1)
+				health = min(health + 20, maxHealth)
+				to_chat(user, "<span class='notice'>You fix some of the dents on \the [src]!</span>")
+			else
+				to_chat(user, "<span class='warning'>Need more welding fuel!</span>")
+				return
+
+	else if(istype(W, /obj/item/device/silicate_sprayer))
+		if(!locked)
+			to_chat(user, "<span class='warning'>You need to lock the shield before you can fix it.</span>")
+		else
+			if(shield >= maxShield)
+				to_chat(user, "<span class='warning'>Nothing to fix here!</span>")
+				return
+			var/obj/item/device/silicate_sprayer/SS = W
+			if(SS.get_amount() >= 5)
+				SS.remove_silicate(5)
+				playsound(loc, 'sound/effects/refill.ogg', 50, 1)
+				shield = min(shield + 20, maxShield)
+				to_chat(user, "<span class='notice'>You fix some of the dents on \the [src]'s shield!</span>")
+			else
+				to_chat(user, "<span class='warning'>Need more silicate!</span>")
+				return
+
+	else if(user.a_intent == I_HURT)
+		user.delayNextAttack(8)
+		getDamage(W.force)
+		user.visible_message("<span class='danger'>[user.name] [W.attack_verb] \the [src]!</span>", "<span class='danger'>You [W.attack_verb] \the [src]!</span>")
+	else
+		return ..()
+
+/obj/structure/mannequin/cyber/update_icon()
+	overlays.len = 0
+	..()
+	if(destroyed)
+		overlays |= image(icon, "mannequin_cover_broken")
+	else if(locked)
+		overlays |= image(icon, "mannequin_cover")
+
+/obj/structure/mannequin/cyber/acidable()
+	return 0
+
+/obj/structure/mannequin/cyber/attack_hand(var/mob/user)
+	if(destroyed)
+		show_inv(user)
+	else if(locked)
+		if(user.a_intent == I_HURT)
+			user.delayNextAttack(8)
+			user.visible_message("<span class='danger'>[user.name] kicks \the [src]!</span>", "<span class='danger'>You kick \the [src]!</span>", "You hear glass crack.")
+			getDamage(2)
+		else
+			to_chat(user,"<span class='notice'>You gently run your hands over \the [src] in appreciation of its contents.</span>")
+	else
+		..()
+
+/obj/structure/mannequin/cyber/examine(mob/user)
+	..()
+	if(destroyed)
+		to_chat(user, "Its glass shield has been shattered.")
+
+/obj/structure/mannequin/cyber/broken
+	health = 100
+	destroyed = 1
+
+
+
+
+
+/////////////////////////////////////////////////////////MANNEQUIN FRAME//////////////////////////////////////////////////////////
+//Used to build cyber mannequins.
+
+/obj/structure/mannequin_frame
+	name = "human cyber mannequin frame"
+	desc = "Lots of work just to display some clothes."
+	icon = 'icons/obj/mannequin_64x64.dmi'
+	icon_state="mannequin_cyber_human_frame"
+	pixel_x = -16
```

Comment:
Should this even be set in code? What's the reason to have manual pixel displacement from the start?